File: ai/src/main/java/org/conductoross/conductor/ai/models/ImageGenRequest.java
package org.conductoross.conductor.ai.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@EqualsAndHashCode(callSuper = true)
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ImageGenRequest extends LLMWorkerInput {
    public enum OutputFormat {
        jpg,
        png,
        webp
    }

    private float weight;
    @Builder.Default private int n = 1;
    private int width = 1024;
    private int height = 1024;
    private String size;
    private String style;
    @Builder.Default private OutputFormat outputFormat = OutputFormat.png;
}

File: ai/src/main/typescript/models/ImageGenRequest.ts
import { LLMWorkerInput } from './LLMWorkerInput.js';

export enum ImageOutputFormat {
  jpg = 'jpg',
  png = 'png',
  webp = 'webp',
}

export class ImageGenRequest extends LLMWorkerInput {
  weight: number = 0;
  n: number = 1;
  width: number = 1024;
  height: number = 1024;
  size?: string;
  style?: string;
  outputFormat: ImageOutputFormat = ImageOutputFormat.png;
}
